src/coreclr/vm/precode.h

```diff
@@ -596,4 +596,39 @@ static_assert_no_msg(NDirectImportPrecode::Type != ThisPtrRetBufPrecode::Type);
 static_assert_no_msg(sizeof(Precode) <= sizeof(NDirectImportPrecode));
 static_assert_no_msg(sizeof(Precode) <= sizeof(FixupPrecode));
 static_assert_no_msg(sizeof(Precode) <= sizeof(ThisPtrRetBufPrecode));
+
+#ifndef DACCESS_COMPILE
+// A summary of the precode layout for diagnostic purposes
+struct PrecodeMachineDescriptor
+{
+    uintptr_t CodePointerToInstrPointerMask;
```

Comment:
I think I want to remove this mask and just add a `CodePointerFlags` enum:

```csharp
[Flags]
enum CodePointerFlags : byte
{
    HasArm32ThumbBit = 0x01,
    HasArm64PtrAuth = 0x02,
}
```

And make the metadata contract provide a `TargetPointer AsTargetPointer(TargetCodePointer)` function that will mask off the thumb bit or get the raw address from an authenticated pointer once that's a problem we have to deal with...